core/dispute_coordinator/impl/dispute_coordinator_impl.cpp

```diff
@@ -999,9 +1005,29 @@ namespace kagome::dispute {
       controlled_indices.emplace(keypair->second);
     }
 
-    return CandidateEnvironment{.session_index = session,
-                                .session = std::move(session_info),
-                                .controlled_indices = controlled_indices};
+    // combine on-chain with off-chain disabled validators
+    // process disabled validators in the following order:
+    // - on-chain disabled validators
+    // - prioritized order of off-chain disabled validators
+    // deduplicate the list and take at most `byzantine_threshold` validators
+    auto byzantine_threshold =
+        (std::max<size_t>(session_info.validators.size(), 1) - 1) / 3;
+    std::unordered_set<ValidatorIndex> disabled_indices;
+    for (auto &container : {disabled_onchain, disabled_offchain}) {
```

Comment:
Use `std::initializer_list<T *>` instead of `std::initializer_list<T>` to avoid copy.
```suggestion
    for (const auto *container : {&disabled_onchain, &disabled_offchain}) {
```